torch/csrc/jit/codegen/cuda/ir_iostream.cpp

```diff
@@ -584,22 +639,40 @@ void IRPrinter::handle(const IfThenElse* ite) {
 
 void IRPrinter::handle(const Allocate* a) {
   indent();
-  os << a->buf_type();
-  if (a->buffer()->getValType() == ValType::TensorView) {
-    os << " T" << a->buffer()->name() << "[";
-    print_inline(a->extent());
-    os << "];\n";
-  } else {
-    if (a->extent()->isOneInt()) {
-      os << " " << a->buffer() << ";\n";
-    } else {
-      TORCH_INTERNAL_ASSERT(
-          false,
-          "Received unexpected allocation: ",
-          a->buffer(),
-          " with alloc of ",
-          a->extent());
+  if (a->buffer()->getValType().value() == ValType::TensorView) {
+    auto tv = a->buffer()->as<TensorView>();
+
+    switch (tv->getMemoryType()) {
+      case (MemoryType::Global):
+        os << "// Allocate global tensor " << a->buffer_type() << " T"
+           << tv->name() << "[";
+        if (a->size() == nullptr) {
+          handle(tv);
+        } else {
+          print_inline(a->size());
+        }
+        os << "];\n";
+        break;
+      case (MemoryType::Shared):
+        os << "__shared__ ";
+        __attribute__((fallthrough));
```

Comment:
I don't think this will compile on Windows. Does pytorch have a portable macro? (or we can pretend we're c++17 and use [[fallthrough]] :)